frontend/document/koptinterface.lua

```diff
@@ -1464,6 +1469,87 @@ function KoptInterface:findText(doc, pattern, origin, reverse, caseInsensitive,
     end
 end
 
+local function get_prev_word(text_boxes, i, j)
+    if j > 1 then -- this line, previous word
+        j = j - 1
+    else
+        if i > 1 then -- previous line, last word
+            i = i - 1
+            j = #text_boxes[i]
+        end
+    end
+    local prev_word = text_boxes[i][j]
+    if prev_word then
+        return prev_word.word, i, j
+    end
+end
+
+local function get_next_word(text_boxes, i, j)
+    if j < #text_boxes[i] then -- this line, next word
+        j = j + 1
+    else
+        if i < #text_boxes then -- next line, first word
+            i = i + 1
+            j = 1
+        end
+    end
+    local next_word = text_boxes[i][j]
+    if next_word then
+        return next_word.word, i, j
+    end
+end
+
+function KoptInterface:findTextAll(doc, pattern, caseInsensitive)
```

Comment:
`findAlltext`?

Reply:
(Unless there's a previous pattern of functions being name `findText<Something>`?)

Reply:
For internal usage there is
https://github.com/koreader/koreader/blob/5d2a4410648dfc993e6dbc01cbf23941a92404c3/frontend/document/koptinterface.lua#L1410
and for external calls
https://github.com/koreader/koreader/blob/5d2a4410648dfc993e6dbc01cbf23941a92404c3/frontend/document/koptinterface.lua#L1430
So, the new external one supposed to be `findTextAll`.